src/driver/discover.rs

```diff
@@ -0,0 +1,24 @@
+use driver::Driver;
+use error::Error;
+use schema;
+
+/// Takes a driver config value and finds and connects the associated driver.
+/// Errors if not driver could be found for the given config.
+// TODO: @svmnotn this is your jam!
+pub fn discover_driver(config: &schema::Driver) -> Result<Box<Driver>, Error> {
+  match config.url().scheme.as_str() {
+    "mongodb" => connect_mongodb_driver(config),
```

Comment:
Look at the definitions for `connect_mongodb_driver` a few lines down. It has two variants which conditionally compile depending on the feature.